rpi_backlight/__init__.py

```diff
@@ -22,14 +23,24 @@ def _permission_denied() -> None:
         "for the backlight access as described in README.md."
     )
 
-
+    
 class Backlight:
     """Main class to access and control the display backlight power and brightness."""
 
     def __init__(
         self, backlight_sysfs_path: Union[str, "PathLike[str]"] = _BACKLIGHT_SYSFS_PATH
     ):
         """Set ``backlight_sysfs_path`` to ``":emulator:"`` to use with rpi-backlight-emulator."""
+        try:
+            with open("/sys/firmware/devicetree/base/model") as f:
+                model_information = f.read()
+            if "Raspberry Pi" in model_information:
+                _BACKLIGHT_SYSFS_PATH = "/sys/class/backlight/rpi_backlight/"
+            elif "Tinker Board" in model_information:
+                _BACKLIGHT_SYSFS_PATH = "/sys/devices/platform/ff150000.i2c/i2c-3/3-0045/"
+        except Exception as error:
+            raise Exception("unsupported OS, or OS could not be detected!")
```

Comment:
This will break usage with github.com/linusg/rpi-backlight-emulator/

Reply:
Can you provide a bit more guidance?
I did not understand where you want any enums or contructors to be edited.

Reply:
sure. I'm thinking something like this:

```python3
from enum import Enum, auto


class BoardType(Enum):
    RASPBERRY_PI = auto()
    TINKER_BOARD = auto()


class Backlight:
    def __init__(
        self,
        backlight_sysfs_path: Union[str, "PathLike[str]"] = _BACKLIGHT_SYSFS_PATH,
        board_type: BoardType = BoardType.RASPBERRY_PI
    ):
        ...
        self.board_type = board_type
```

and then for example

```python3
    @power.setter
    def power(self, on: bool) -> None:
        """Set the display power on or off."""
        if not isinstance(on, bool):
            raise TypeError("value must be a bool, got {0}".format(type(on)))

        if self.board_type == BoardType.RASPBERRY_PI:
            # 0 is on, 1 is off
            self._set_value("bl_power", int(not on))
        elif self.board_type == BoardType.TINKER_BOARD:
            # Tinker Board specific code
```

This would mean you can continue to use `Backlight()` without any changes on a Raspberry Pi and `Backlight(board_type=BoardType.TINKER_BOARD)` on the Tinker Board.

Does this help?

Reply:
@linusg yes, that helps, thanks. 
How do you want me to detect the OS and change the file path based on that?
Or you are thinking that the end user will enter manually edit the code as such:
```
board_type: BoardType = BoardType.TINKER_BOARD
```

Reply:
```
_BACKLIGHT_SYSFS_PATH_TINKERBOARD = "/sys/devices/platform/ff150000.i2c/i2c-3/3-0045/"
```
So just end user will replace in the code (so no OS auto detection?):
```
backlight_sysfs_path: Union[str, "PathLike[str]"] = _BACKLIGHT_SYSFS_PATH_TINKERBOARD,
        board_type: BoardType = BoardType.TINKER_BOARD
```

Reply:
@linusg Hi, yes, that helps very much.
I've been working on this for 2 hours so far and seem to hit a roadblock. 
First, I'd like to check in with you and make sure I am following your plan so far.
Second, I can't seem to get this to work (line 54 + 190 of init)
Can you please refer to https://github.com/p1r473/rpi-backlight/tree/master/rpi_backlight (so I dont have to do a new pull request to show you the changes as I closed this one)
```
if((self._board_type == BoardType.TINKER_BOARD)
```
Can you see if I am doing something bad with enum?

BTW, I couldnt use auto for enum, I think it is not supported on my Tinkerboard... Was not able to import it.